code/referenceAndAuto/referenceAndAuto.cpp

```diff
@@ -0,0 +1,48 @@
+#include <cstdio>
+
+struct DontCopyMe {
+   int resultA;
+   int resultB;
+
+   // Again, material for the second day:
+   DontCopyMe() = default;
+   DontCopyMe(const DontCopyMe& other) :
+       resultA(other.resultA),
+       resultB(other.resultB)
+   { printf("Please don't copy me\n"); }
+};
+
+int main() {
+   // We create an array of DontCopyMe structs:
+   DontCopyMe collection[10];
+
+   // Task 1:
+   // Write a for loop that initialises each struct's resultA and resultB with ascending integers.
+   // Verify the output of the program before and after you do this.
+
+
+   // Task 2:
+   // We use a range-based for loop to analyse the array of structs.
+   // The problem is: we are copying every DontCopyMe ...
+   // Fix this loop using references.
+   // Hint: Fix the type declaration "auto" in the loop head.
+   int resultA = 0;
+   int resultB = 0;
+   for (auto item : collection) {
+      resultA += item.resultA;
+      resultB += item.resultB;
+   }
+
+
+   printf("resultA = %d\tresultB = %d\n", resultA, resultB);
+
+   return 0;
+}
+
+// Task 3:
+// Make sure that your references/pointers are const if you don't need to modify the data
+// in DontCopyMe.
+// Hint: C++ understands "const auto".
+
+// Task 4 (optional):
+// Repeat the above loop using a range-based for loop with pointers to the elements.
```

Comment:
Not sure I understand the task here. Can you be more precise ?

Reply:
I was initially thinking about a collection of pointers. This doesn't make sense here, so I remove it now.